frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -188,8 +188,10 @@ export const PublicFormProvider = ({
 
   const { handleLogoutMutation } = usePublicAuthMutations(formId)
 
-  const handleSubmitForm: SubmitHandler<FormFieldValues> = useCallback(
-    async (formInputs) => {
+  const handleSubmitForm: SubmitHandler<
+    FormFieldValues & { payment_receipt_email_field?: { value: string } }
+  > = useCallback(
+    async ({ payment_receipt_email_field, ...formInputs }) => {
```

Comment:
This type feels a bit strange to me. For one, the `payment_receipt_email_field` does not occur in all form submissions, only the ones with payments. Additionally, the field is not explicitly defined as a form field, so i'm not sure that it is most intuitive for the submission handler to take it as input. 

Maybe one way to get around this is to create a ref for the email input within the public form context, then we can access the value of the ref when we need it. Another alternative is we could create a state variable within the public form context and set the state on the input.

What do you think? We can discuss this further!

Reply:
Hm but if we were to use a ref or state in the public form context, wouldn’t that mean that those additional resources will have to be created for all forms whether or not payment is added to the particular form?

Reply:
Outcome from 15 Mar sync: treat it as separate with refs/states, or be more explicit with the typing.

Since we have decided on having verification on the email field as well, and we should reuse the verification flow for regular email inputs, it might make more sense to inject the field into the form rather than a separate ref/state since the verification flow is tied to fieldId, which is used in the DB to save the hashed OTP too. I'm still trying to figure out how the verification flow works 😅

Reply:
Okay we'll relook at this in a separate issue + PR, for now, considerations are documented [here](https://docs.google.com/document/d/1Js4zR8guyT-gREK4yTC094UZlJ96zRELTVgiviFKDEM/edit#heading=h.gmpenlmawtn1).